ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -54,18 +57,29 @@ struct S3_STORAGE {
 
     uint32_t referenceCount; /* Number of references to this storge source */
     int32_t verbose;
+
+    std::list<S3_FILE_HANDLE *> fhList;
```

Comment:
I prefer we bunch the list and the mutex for filehandle with the file system in one block. Also please add comments on what the mutex and list is for:

```
/* S3 storage source structure. */
struct S3_STORAGE {
    WT_STORAGE_SOURCE storageSource; /* Must come first */
    WT_EXTENSION_API *wtApi;         /* Extension API */

    std::mutex fsListMutex;             /* Protect the file system list */
    std::list<S3_FILE_SYSTEM *> fsList; /* List of initiated file systems */
    std::mutex fhMutex;                  /* Protect the file handle list */
    std::list<S3_FILE_HANDLE *> fhList; /* List of active file handles */

    uint32_t referenceCount; /* Number of references to this storge source */
    int32_t verbose;
};
```